cinch/playbooks/jenkins_restart_utility.yml

```diff
@@ -0,0 +1,139 @@
+              copy:
+                content: "{{ thread_dump.stdout }}"
+                dest: "{{ diag_path }}/jstack.txt"
+              when: diag_path_creation|success
+
+            # TODO: add the following package so that this task can work properly:
+            # java-1.8.0-openjdk-debuginfo
+            - name: generate JVM memory report (jmap)
+              command:
+                timeout 60 {{ jmap_path | default('/usr/bin/jmap') }} -heap {{ jenkins_pid }}
+              register: jvm_memory_jmap
+
+            - name: copy JVM memory report (jmap) to diagnostics directory
+              copy:
+                content: "{{ jvm_memory_jmap.stdout }}"
+                dest: "{{ diag_path }}/jvm-jmap.txt"
+              when: diag_path_creation|success
+
+            - name: generate JVM memory report (jstat)
+              command:
+                timeout 60 {{ jstat_path | default('/usr/bin/jstat') }} -gcutil {{ jenkins_pid }}
+              register: jvm_memory_jstat
+
+            - name: copy JVM memory report (jstat) to diagnostics directory
+              copy:
+                content: "{{ jvm_memory_jstat.stdout }}"
+                dest: "{{ diag_path }}/jvm-jstat.txt"
+              when: diag_path_creation|success
+
+            # By default, the 'root' Unix user account is assumed to have the
+            # necessary SSH key to authenticate to Jenkins as an admin user who has
+            # the authorization to generate the support bundle.
+            - name: generate support bundle
+              command: >
+                timeout 120 {{ java_path | default('/usr/bin/java') }} -jar
+                {{ jenkins_cli_path | default('/var/cache/jenkins/war/WEB-INF/jenkins-cli.jar') }}
+                -s http://localhost:8080 support
+              register: support_bundle
+              become: true
+              become_user: "{{ jenkins_cli_user | default('root') }}"
+
+            # Strangely, the Jenkins CLI reports the necessary output for a
+            # successful run of the support bundle on stderr
+            - name: find support bundle file
+              set_fact:
+                support_bundle_file:
+                  "{{ support_bundle.stderr | regex_replace('^Creating: ', '') }}"
+              when: support_bundle.rc == 0
+
+            - name: copy support bundle file to diagnostics directory
+              copy:
+                src: "{{ support_bundle_file }}"
+                dest: "{{ diag_path }}"
+                remote_src: true
+              when: diag_path_creation|success and support_bundle_file is defined
+          when: jenkins_running.stat.isreg is defined
+          ignore_errors: true
+      become: true
+      become_user: jenkins
+
+    # always restart Jenkins if restart_jenkins is true
+    - name: restart Jenkins
+      service:
+        name: jenkins
+        state: restarted
+      when: restart_jenkins | default(false) and ansible_connection != 'docker'
```

Comment:
It is my personal preference to wrap parenthesis around uses of filters when these only affect part of the expression. Nowhere have I seen the order of operations for filter evaluation vs Boolean algebra defined, and wrapping the parenthesis just gives us a clearer notion that the "and" is connected to the whole term on the left

``when: ( restart_jenkins | default(false) ) and ansible_connection != 'docker'``